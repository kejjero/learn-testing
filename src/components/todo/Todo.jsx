import Task from "../task/Task";
import {useState} from "react";

function Todo({todos, setTodos}) {


    const [value, setValue] = useState('');

    function onClickAddTodo(evt) {
        evt.preventDefault();
        setTodos([...todos, value])
        setValue('')
    }

    function onClickDeleteTodo(indexTodo) {
        const newTodo = todos.filter((_, currentIndex) => currentIndex !== indexTodo);
        setTodos(newTodo)

    }

    if (!todos) return null

    return (
        <div className="todo">
            <ul>
                {
                    todos.map((todo, index) => {
                        return (
                            <Task
                                key={index}
                                index={index}
                                todo={todo}
                                onClickDeleteTodo={onClickDeleteTodo}
                            />
                        )
                    })
                }
            </ul>
            <form className="todo__form">
                <input type="text" value={value} onChange={(evt) => setValue(evt.target.value)}/>
                <button
                    onClick={(evt) => onClickAddTodo(evt)}
                >
                    Добавить задачу
                </button>
            </form>
        </div>
    )
}

export default Todo;