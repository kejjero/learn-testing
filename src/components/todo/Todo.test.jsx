import {render, screen} from '@testing-library/react';
import Todo from '../todo/Todo';


describe('Todo компонент', () => {

    const todos = ['todo one', 'todo two', 'todo three']

    it('Создание компонента Todo с данными', () => {

        render(<Todo todos={todos}/>);

        const todoGetRole = screen.queryByRole('list');
        const todoGetText = screen.getByText(/todo three/i);

        expect(todoGetRole).toMatchSnapshot()
        expect(todoGetRole).toBeInTheDocument();
        expect(todoGetText).toBeInTheDocument();
    });

    it('Создание компонента без данных', () => {

        render(<Todo/>);

        const todoGetRole = screen.queryByRole('list');

        expect(todoGetRole).toBeNull()
    });

});