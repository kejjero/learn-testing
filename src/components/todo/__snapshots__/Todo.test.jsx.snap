// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo компонент  1`] = `
<ul>
  <li>
    <button>
      x
    </button>
    <p>
      todo one
    </p>
  </li>
  <li>
    <button>
      x
    </button>
    <p>
      todo two
    </p>
  </li>
  <li>
    <button>
      x
    </button>
    <p>
      todo three
    </p>
  </li>
</ul>
`;

exports[`Todo компонент Проверка создания копонента Todo с данными 1`] = `
<ul>
  <li>
    <button>
      x
    </button>
    <p>
      todo one
    </p>
  </li>
  <li>
    <button>
      x
    </button>
    <p>
      todo two
    </p>
  </li>
  <li>
    <button>
      x
    </button>
    <p>
      todo three
    </p>
  </li>
</ul>
`;
